app/controllers/api/comments_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Api::CommentsController < Api::ApplicationController
+  def index
+    @post = Post.find(params[:post_id])
+    if @post
+      render json: { status: 'ok', data: @post.comments }
+    else
+      render json: { status: 'Error', error: 'Post Not Found' }
+    end
+  end
+
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new(comment_params)
+    @comment.author = current_user
+    @comment.post = @user.posts.find(params['post_id'])
+
+    if @comment.save
+      render json: { status: :created, data: @comment }
+    else
+      render json: { error: @comment.errors, status: :unprocessable_entity }
```

Comment:
- [ ] You're finding the user and post by ID, but you're not checking whether they actually exist or not. It's a good practice to add some error handling for cases where the user or post is not found, to provide a more helpful error message to the API client. :wink: KIndly fix this